### PR TITLE
Bugfix lesson card questions

### PR DESCRIPTION
**Pull request by ShevchenkoKirilo**
Fixed bugs 166, 216, 272

### CODE REVIEW

**Review by KaterynaHerasymiva — DISMISSED** [review]

File: frontend/src/app/modules/lessons/lesson/lesson.component.ts [hunk 1]
```diff
@@ -57,12 +61,27 @@ export class LessonComponent extends BaseComponent implements OnInit {
...
-        if (this.questions) {
-            this.isShowQuestions = !this.isShowQuestions;
-        }
-
-        if (this.isShowQuestions) {
+        if (this.questions && this.isQuestionsOpened) {
+            this.questions = '';
+            this.isQuestionsOpened = false;
+        } else {
+            this.isQuestionsOpened = true;
```
Comment (tatianahutii) [inline frontend/src/app/modules/lessons/lesson/lesson.component.ts]: ternary operator

File: frontend/src/app/modules/lessons/lesson/lesson.component.ts [hunk 2]
```diff
@@ -57,12 +61,27 @@ export class LessonComponent extends BaseComponent implements OnInit {
...
+                if (container) {
+                    observer.disconnect();
+
+                    if (container.scrollHeight > container.clientHeight) {
+                        container.style.height = '460px';
+                    }
+                }
+            });
+
+            observer.observe(document.body, { childList: true, subtree: true });
```
Comment (tatianahutii) [inline frontend/src/app/modules/lessons/lesson/lesson.component.ts]: separate method

File: frontend/src/app/modules/lessons/lesson/lesson.component.ts [hunk 3]
```diff
@@ -98,10 +117,12 @@ export class LessonComponent extends BaseComponent implements OnInit {
 
     isDisabled() {
         if (!this.isTeachersPage) {
-            return this.lesson.isSubscribed
-                || new Date() > new Date(this.lesson.startAt)
-                || this.lesson.limitOfUsers === null
-                || this.lesson.subscribersCount === this.lesson.limitOfUsers;
+            return (
+                this.lesson.isSubscribed ||
+                new Date() > new Date(this.lesson.startAt) ||
```
Comment (tatianahutii) [inline frontend/src/app/modules/lessons/lesson/lesson.component.ts:124]: duplication

File: frontend/src/app/modules/lessons/lessons-page/lessons-page.component.html [hunk 4]
```diff
@@ -5,13 +5,21 @@ <h2 class="date">{{ todayDate }}</h2>
...
         </div>
         <div class="lessons-column">
-            <div class="lesson" *ngFor="let lesson of lessonsColumn2">
-                <app-lesson [lesson]="lesson" [isTeachersPage]="false"></app-lesson>
+            <div class="lesson" *ngFor="let lesson of lessonsColumn2; let i = index">
+                <app-lesson
+                    [lesson]="lesson"
+                    [isTeachersPage]="false"
+                    [isQuestionsOpened]="isLessonOpenedQuestions(i + lessonsColumn1.length)"
+                    (questionsOpened)="onLessonOpenedQuestions(i + lessonsColumn1.length)"></app-lesson>
```
Comment (tatianahutii) [inline frontend/src/app/modules/lessons/lessons-page/lessons-page.component.html]: looks a little bit strange 
try to refactor these lines

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Review by KaterynaHerasymiva — DISMISSED** [review]

File: frontend/src/app/modules/user/teachers-page/teachers-page.component.html [hunk 5]
```diff
@@ -66,23 +66,32 @@ <h1>Teachers Profile</h1>
...
+                        [lesson]="lesson"
+                        [isTeachersPage]="true"
+                        [isQuestionsOpened]="isLessonOpenedQuestions(lesson.id)"
+                        (questionsOpened)="onLessonOpenedQuestions(lesson.id)"></app-lesson>
                 </div>
             </div>
             <div class="lessons-column">
-                <div class="lesson" *ngFor="let lesson of day.lessonsColumn2">
-                    <app-lesson [lesson]="lesson" [isTeachersPage]="true"></app-lesson>
+                <div class="lesson" *ngFor="let lesson of day.lessonsColumn2; let j = index">
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/user/teachers-page/teachers-page.component.html]: I guess you don't need indexes anymore

File: frontend/src/app/modules/lessons/lessons-page/lessons-page.component.html [hunk 6]
```diff
@@ -5,13 +5,21 @@ <h2 class="date">{{ todayDate }}</h2>
...
+                    [lesson]="lesson"
+                    [isTeachersPage]="false"
+                    [isQuestionsOpened]="isLessonOpenedQuestions(lesson.id)"
+                    (questionsOpened)="onLessonOpenedQuestions(lesson.id)"></app-lesson>
             </div>
         </div>
         <div class="lessons-column">
-            <div class="lesson" *ngFor="let lesson of lessonsColumn2">
-                <app-lesson [lesson]="lesson" [isTeachersPage]="false"></app-lesson>
+            <div class="lesson" *ngFor="let lesson of lessonsColumn2; let i = index">
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/lessons/lessons-page/lessons-page.component.html]: The same here

**Pull request merged**